go/logic/applier.go

```diff
@@ -934,6 +937,42 @@ func (this *Applier) CreateAtomicCutOverSentryTable() error {
 	return nil
 }
 
+// WaitForAtomicCutOverRename waits for the cut-over RENAME operation while periodically
+// pinging the applier connection to avoid the connection becoming idle.
+func (this *Applier) WaitForAtomicCutOverRename(tx *gosql.Tx, okToUnlockTable <-chan bool) {
```

Comment:
I'm not sure this function is required, and it adds more complexity. If we set `@@session.wait_timeout` to be somewhat bigger than lock-wait-timeout, then we don't need to do a connection keep-alive. I see the idea behind this, but I think it's simpler to just set an appropriate `wait_timeout` and let MySQL handle it.

Reply:
@shlomi-noach see earlier comment. This is born by paranoia that the `RENAME` in the other session could take longer than `lock_wait_timeout`, which I believe is possible (eg: server is slow, network, process scheduling, hand-wavy things, etc)

To be fair, these edge cases are extremely "edge"

Reply:
I'd say I still think this function adds complexity, and still better to use `@@wait_timeout` based on some (same) formula such as the one presented above.